libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1167,6 +1167,30 @@ bool AP_Arming::pre_arm_checks(bool report)
 
 bool AP_Arming::arm_checks(AP_Arming::Method method)
 {
+    if (method == AP_Arming::Method::RUDDER) {
+        if (get_rudder_arming_type() == AP_Arming::RudderArming::IS_DISABLED) {
```

Comment:
How do we have a code path where we issue the arm, but the type is disabled? That seems like something we'd want a pretty strong internal error on, if not an assert....

Reply:
This helps centralise the arming checks.  This means we don't need to check if the type of arming is disabled in the caller - we've got the type being attempted already being passed in here for a reason...